server_addon/houdini/client/ayon_houdini/plugins/publish/collect_files_for_cleaning_up.py

```diff
@@ -0,0 +1,115 @@
+import os
+
+import clique
+import pyblish.api
+from ayon_core.pipeline import AYONPyblishPluginMixin
+from ayon_houdini.api import lib, plugin
+
+
+class CollectFilesForCleaningUp(plugin.HoudiniInstancePlugin,
+                                AYONPyblishPluginMixin):
+    """Collect Files For Cleaning Up.
+
+    This collector collects output files
+    and adds them to file remove list.
+
+    CAUTION:
+        This collector deletes the exported files and
+          deletes the parent folder if it was empty.
+        Artists are free to change the file path in the ROP node.
+    """
+
+    # It should run after CollectFrames and Collect Render plugins,
+    # and before CollectLocalRenderInstances.
+    order = pyblish.api.CollectorOrder + 0.115
+
+    hosts = ["houdini"]
+    families = ["*"]
+    label = "Collect Files For Cleaning Up"
+    intermediate_exported_render = False
+
+    def process(self, instance):
+        import hou  # noqa: E402
+
+        node = hou.node(instance.data.get("instance_node", ""))
+        if not node:
+            self.log.debug(
+                "Skipping Collector. Instance has no instance_node")
+            return
+
+        output_parm = lib.get_output_parameter(node)
+        if not output_parm:
+            self.log.debug(
+                f"ROP node type '{node.type().name()}' is not "
+                "supported for cleaning up.")
+            return
+
+        filepath = output_parm.eval()
+        if not filepath:
+            self.log.warning("No filepath value to collect.")
+            return
+
+        files = []
+        staging_dir, _ = os.path.split(filepath)
+
+        expected_files = instance.data.get("expectedFiles", [])
+
+        # 'expectedFiles' are preferred over 'frames'
+        if expected_files:
+            # Products with expected files
+            # This can be Render products or submitted cache to farm.
+            for expected in expected_files:
+                # expected.values() is a list of lists
+                files.extend(sum(expected.values(), []))
```

Comment:
I understand what this is doing yet it still looks weird - you seem to want to flatten or concatenate the list into one long list.

But why not just:
```suggestion
                for output_files in expected.values():
                    files.extend(output_files)
```
_Or whatever is a good variable name_. It seems less 'magic' that way?